src/System.Security.Principal.Windows/tests/WindowsIdentityTests.cs

```diff
@@ -28,44 +30,58 @@ public static void GetAnonymousUserTest()
     [Fact]
     public static void ConstructorsAndProperties()
     {
-        // Retrieve the Windows account token for the current user.
-        SafeAccessTokenHandle token = WindowsIdentity.GetCurrent().AccessToken;
-        bool gotRef = false;
-        try
+        TestUsingAccessToken((logonToken) =>
         {
-            token.DangerousAddRef(ref gotRef);
-            IntPtr logonToken = token.DangerousGetHandle();
-
             // Construct a WindowsIdentity object using the input account token.
-            WindowsIdentity windowsIdentity = new WindowsIdentity(logonToken);
+            var windowsIdentity = new WindowsIdentity(logonToken);
             Assert.NotNull(windowsIdentity);
             CheckDispose(windowsIdentity);
 
-            string authenticationType = "WindowsAuthentication";
-            WindowsIdentity windowsIdentity2 = new WindowsIdentity(logonToken, authenticationType);
+            var windowsIdentity2 = new WindowsIdentity(logonToken, authenticationType);
             Assert.NotNull(windowsIdentity2);
             Assert.True(windowsIdentity2.IsAuthenticated);
 
             Assert.Equal(authenticationType, windowsIdentity2.AuthenticationType);
             CheckDispose(windowsIdentity2);
-        }
-        finally
-        {
-            if (gotRef)
-                token.DangerousRelease();
-        }
+        });
     }
 
     [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public static void AuthenticationCtor(bool authentication)
+    {
+        TestUsingAccessToken((logonToken) =>
+        {
+            var windowsIdentity = new WindowsIdentity(logonToken, authenticationType, WindowsAccountType.Normal, isAuthenticated: authentication);
+            Assert.NotNull(windowsIdentity);
```

Comment:
Constructors in .NET cannot return null. You do not need asserts like this.

Reply:
I see there are some elsewhere in the file..

Reply:
I deleted all of them